crates/re_viewer/src/app_state.rs

```diff
@@ -105,12 +105,18 @@ impl AppState {
             viewport_state,
         } = self;
 
+        // Some of the mutations APIs of `ViewportBlueprints` are recorded as `Viewport::TreeAction`
+        // and must be applied by `Viewport` at the end of the frame. We use a temporary channel for
+        // this, which gives us interior mutability (only a shared reference of `ViewportBlueprint`
+        // is available to the UI code) and, if needed in the future, concurrency.
+        let (sender, receiver) = std::sync::mpsc::channel();
```

Comment:
Creating a ad-hoc channel on every frame feels a little bit weird, but it does the job. One alternative would be to store the channel in `AppState`, but this introduce alternative weirdnesses (such as "state" containing channel) and small annoyances (introduce a lifetime generic arg to `ViewportBlueprint` to store a ref to the the sender).

Reply:
I guess the sender could be cloned each time, which would spare the lifetime arg for `ViewportBlureprint`.

Reply:
Is the overhead actually any higher than a mutex + vec?

Reply:
I dont see a reason it would be. I actually am fine with it this way, I just wanted to shine some light on it to be on the safe side.